src/middleware/live-middleware.ts

```diff
@@ -49,9 +49,11 @@ export class KalturaLiveMiddleware extends KalturaPlayer.core.BaseMiddleware {
 
     public load(next: Function): void {
         // if plugin is not active (E.G. in VOD) the middleware will not work
-        if (!this._livePlugin.isLiveMediaLoaded) {
+        if (!this._livePlugin.isMediaLive) {
```

Comment:
LGTM but I have to say how it wrote today it's super hard to understand the logic, for load and play
I suggest simplify it, load method will look like this:
```
if (this._livePlugin.isMediaLive && !this.isPlayerLive()) {
    this._livePlugin.updateLiveStatus();
    this._nextLoad = next;
    this.initialPlayHandling();
    logger.info("interrupt load", { method: "load" });
} else {
    this.callNext(next);
}
```